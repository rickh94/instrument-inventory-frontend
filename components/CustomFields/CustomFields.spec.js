import React from 'react'
import {
  InstrumentTypeSelect,
  InstrumentSizeSelect,
  LocationSelect,
} from './CustomFields'
import 'jest-dom/extend-expect'
import { render, fireEvent } from '@testing-library/react'

describe('Fields', () => {
  describe('InstrumentTypeSelect', () => {
    it('matches snapshot', () => {
      const { container } = render(
        <InstrumentTypeSelect
          error={{}}
          value={'violin'}
          onChange={jest.fn()}
          required={false}
        />
      )
      expect(container).toMatchSnapshot()
    })

    it('calls function on change', () => {
      const onChangeMock = jest.fn()
      const { container } = render(
        <InstrumentTypeSelect
          error={{}}
          value={'violin'}
          onChange={onChangeMock}
          required={false}
        />
      )

      fireEvent.change(container.querySelector('select#instrument-type'), {
        target: { value: 'viola' },
      })
      expect(onChangeMock).toHaveBeenCalled()
    })

    it('does not set required', () => {
      const { container } = render(
        <InstrumentTypeSelect error={{}} value={'violin'} onChange={jest.fn()} />
      )

      expect(container.querySelector('select#instrument-type')).not.toHaveAttribute(
        'required'
      )
    })

    it('does set required', () => {
      const { container } = render(
        <InstrumentTypeSelect
          error={{}}
          value={'violin'}
          onChange={jest.fn()}
          required
        />
      )

      expect(container.querySelector('select#instrument-type')).toHaveAttribute(
        'required'
      )
    })
  })

  describe('InstrumentSizeSelect', () => {
    it('matches snapshot', () => {
      const { container } = render(
        <InstrumentSizeSelect
          error={{}}
          value="4/4"
          onChange={jest.fn()}
          required={false}
          instrumentType="violin"
        />
      )
      expect(container).toMatchSnapshot()
    })

    it('renders different sizes for viola', () => {
      const { queryByText } = render(
        <InstrumentSizeSelect
          error={{}}
          value="4/4"
          onChange={jest.fn()}
          required={false}
          instrumentType="viola"
        />
      )
      expect(queryByText('12"').length).not.toBe(0)
      expect(queryByText('4/4').length).toBeFalsy()
    })

    it('calls function on change', () => {
      const onChangeMock = jest.fn()
      const { container } = render(
        <InstrumentSizeSelect
          error={{}}
          value="4/4"
          onChange={onChangeMock}
          required={false}
          instrumentType="violin"
        />
      )

      fireEvent.change(container.querySelector('select#size'), {
        target: { value: '1/2' },
      })
      expect(onChangeMock).toHaveBeenCalled()
    })

    it('does not set required', () => {
      const { container } = render(
        <InstrumentSizeSelect
          error={{}}
          value="4/4"
          instrumentType="violin"
          onChange={jest.fn()}
        />
      )

      expect(container.querySelector('select#size')).not.toHaveAttribute('required')
    })

    it('does set required', () => {
      const { container } = render(
        <InstrumentSizeSelect
          error={{}}
          value="4/4"
          onChange={jest.fn()}
          instrumentType="violin"
          required
        />
      )

      expect(container.querySelector('select#size')).toHaveAttribute('required')
    })
  })

  describe('LocationSelect', () => {
    it('matches snapshot', () => {
      const { container } = render(
        <LocationSelect
          error={{}}
          value="Grant Elementary School"
          onChange={jest.fn()}
          required={false}
        />
      )
      expect(container).toMatchSnapshot()
    })

    it('calls function on change', () => {
      const onChangeMock = jest.fn()
      const { container } = render(
        <LocationSelect
          error={{}}
          value=""
          onChange={onChangeMock}
          required={false}
        />
      )

      fireEvent.change(container.querySelector('select#location'), {
        target: { value: 'Trenton High School' },
      })
      expect(onChangeMock).toHaveBeenCalled()
    })

    it('does not set required', () => {
      const { container } = render(
        <LocationSelect error={{}} value="trade" onChange={jest.fn()} />
      )

      expect(container.querySelector('select#location')).not.toHaveAttribute(
        'required'
      )
    })

    it('does set required', () => {
      const { container } = render(
        <LocationSelect
          error={{}}
          value=""
          onChange={jest.fn()}
          required
        />
      )

      expect(container.querySelector('select#location')).toHaveAttribute(
        'required'
      )
    })
  })
})
