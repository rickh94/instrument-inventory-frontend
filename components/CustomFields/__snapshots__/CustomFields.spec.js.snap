// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields InstrumentSizeSelect matches snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root Mui-error Mui-error MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
      data-shrink="true"
      for="size"
    >
      Size
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl"
    >
      <div
        class="MuiNativeSelect-root"
      >
        <select
          aria-invalid="true"
          class="MuiNativeSelect-select MuiInputBase-input MuiInput-input"
          id="size"
        >
          <option
            value=""
          />
          <option
            value="4/4"
          >
            4/4
          </option>
          <option
            value="3/4"
          >
            3/4
          </option>
          <option
            value="1/2"
          >
            1/2
          </option>
          <option
            value="1/4"
          >
            1/4
          </option>
          <option
            value="1/8"
          >
            1/8
          </option>
          <option
            value="1/10"
          >
            1/10
          </option>
          <option
            value="1/16"
          >
            1/16
          </option>
          <option
            value="1/32"
          >
            1/32
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiNativeSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Fields InstrumentTypeSelect matches snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root Mui-error Mui-error MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
      data-shrink="true"
      for="instrument-type"
    >
      Instrument Type
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl"
    >
      <div
        class="MuiNativeSelect-root"
      >
        <select
          aria-invalid="true"
          class="MuiNativeSelect-select MuiInputBase-input MuiInput-input"
          id="instrument-type"
        >
          <option
            value=""
          />
          <option
            value="violin"
          >
            Violin
          </option>
          <option
            value="viola"
          >
            Viola
          </option>
          <option
            value="cello"
          >
            Cello
          </option>
          <option
            value="bass"
          >
            Bass
          </option>
          <option
            value="violin strug as viola"
          >
            Violin strung as viola
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiNativeSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Fields LocationSelect matches snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root Mui-error Mui-error MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
      data-shrink="true"
      for="location"
    >
      Location
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl"
    >
      <div
        class="MuiNativeSelect-root"
      >
        <select
          aria-invalid="true"
          class="MuiNativeSelect-select MuiInputBase-input MuiInput-input"
          id="location"
        >
          <option
            value=""
          />
          <option
            value="Grant Elementary School"
          >
            Grant Elementary School
          </option>
          <option
            value="Hedgepath Middle School"
          >
            Hedgepath Middle School
          </option>
          <option
            value="Wilson Elementary School"
          >
            Wilson Elementary School
          </option>
          <option
            value="Trenton High School"
          >
            Trenton High School
          </option>
          <option
            value="Columbus Elementary School"
          >
            Columbus Elementary School
          </option>
          <option
            value="office"
          >
            Office
          </option>
          <option
            value="Storage"
          >
            Storage
          </option>
          <option
            value="trade"
          >
            trade
          </option>
          <option
            value="maintenance"
          >
            Maintenance
          </option>
          <option
            value="transit"
          >
            Transit
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiNativeSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
